gene_benchmark/tasks.py

```diff
@@ -88,6 +88,19 @@ def _convert_df_to_3d_array(df: pd.DataFrame):
     return np.hstack(reshaped_embeddings)
 
 
+def _get_none_nan_instancies(
+    encodings: pd.DataFrame, outcomes: pd.Series | pd.DataFrame
+):
+    if encodings.shape[1] == 1:
+        nan_ind = encodings.squeeze().isna()
+    else:
+        nan_ind = encodings.isna().any()
+    if isinstance(outcomes, pd.Series):
+        return outcomes[~nan_ind]
+    else:
+        return outcomes.loc[~nan_ind, :]
+
```

Comment:
also here do we really need this if statement? both series and dataframes support boolean indexing so this should work: 
`outcomes.loc[~nan_mask]`

Reply:
so in overall the function: 
`nan_rows = df.isna().any(axis=1)
 return outcomes.loc[~nan_rows]`

Reply:
You are right